### PR TITLE
Add EnzymeCore changes for split mode

### PR DESCRIPTION
**Pull request by vchuravy**
@wsmoses what do you think about this as an API?

### CODE REVIEW

**Comment by wsmoses** [comment]
@vchuravy how would you get the augmented fwd as distinct from gradient in this case?

**Comment by vchuravy** [comment]
So in #245 I would evaluate the augmented primal immediatly and then return the gradient thunk to the user.

It might generally be more flexible to provide as a base API just one that returns the thunks.

**Comment by wsmoses** [comment]
Okay, I'm okay with this. The other two things we may want to include in this:
* whether or not to return the original return.
* whether or not to free in the reverse (or forward split).

**Review by wsmoses — APPROVED** [review]

File: src/Enzyme.jl [hunk 1]
```diff
@@ -21,12 +21,15 @@ export EnzymeRules
...
 import LLVM
 include("api.jl")
 
+convert(::Type{API.CDerivativeMode}, ::ReverseMode{false}) = API.DEM_ReverseModeCombined
+convert(::Type{API.CDerivativeMode}, ::ReverseMode{true}) = API.DEM_ReverseModeGradient
+convert(::Type{API.CDerivativeMode}, ::ForwardMode) = API.DEM_ForwardMode
+
+guess_activity(::Type{T}, mode::Mode) where T = guess_activity(T, convert(API.CDerivativeMode, mode))
+
 @inline function guess_activity(::Type{T}, Mode::API.CDerivativeMode=API.DEM_ReverseModeCombined) where {T}
```
Comment (vchuravy) [inline src/Enzyme.jl:33]:
```suggestion
@inline function guess_activity(::Type{T}, Mode::API.CDerivativeMode) where {T}
```

**Pull request merged**